### DISCUSSION TITLE
Add "Add to Chat Context" Option in Right-Click Menu for Open Files

### DISCUSSION THREAD

**Discussion by Oliver84** [cline/cline#563] [Feature Requests]
I'd love to be able to quickly add the contents of a file to the current chat's context.

Add a new option to the right-click menu for open files in Cline: "Add to Chat Context". When selected, this option should add the entire contents of the file to the current chat's context.

**Comment by dcbartlett** [comment]
The contents of a file can be added using `@<file>`.  This functionality should solve this.

**Reply by dcbartlett** [reply]
@y0m0 I'm not sure why you down-voted this.

**Reply by y0m0** [reply]
Because it's not really the same. When using @file you have to manually write the name of the file that you want to include.
From my understanding what is being suggested here is to add an action, or a right click option to add the file reference.
Kind of like what Copilot already does
![image](https://github.com/user-attachments/assets/730af814-4e44-4a40-b9b7-281b210570c0)
